TeachingRecordSystem/src/TeachingRecordSystem.Core/ReferenceDataCache.cs

```diff
@@ -1,10 +1,15 @@
+using TeachingRecordSystem.Core.DataStore.Postgres;
+using TeachingRecordSystem.Core.DataStore.Postgres.Models;
+using TeachingRecordSystem.Core.Dqt;
 using TeachingRecordSystem.Core.Dqt.Queries;
 
-namespace TeachingRecordSystem.Core.Dqt;
+namespace TeachingRecordSystem.Core;
 
-public class ReferenceDataCache : IStartupTask
+public class ReferenceDataCache(
+    ICrmQueryDispatcher crmQueryDispatcher,
+    TrsDbContext dbContext) : IStartupTask
```

Comment:
Think this’ll need to be `IDbContextFactory` as this type is used as a singleton.